pyanaconda/core/util.py

```diff
@@ -217,15 +217,19 @@ def startX(argv, output_redirect=None, timeout=X_TIMEOUT):
         :param output_redirect: file or file descriptor to redirect stdout and stderr to
         :param timeout: Number of seconds to timing out.
     """
-    # Use a list so the value can be modified from the handler function
-
+    # Use a class so the values can be modified from the handler functions
     class X11Status:
         def __init__(self):
             self.started = False
+            self.timed_out = False
+
+        def needs_waiting(self):
+            return not (self.started or self.timed_out)
```

Comment:
Please verify the choice of data structure. (I'm sure `Enum` or `namedtuple` would not help anything, though.)